src/js/contentscript.js

```diff
@@ -67,6 +68,28 @@ window.addEventListener('message', (event) => {
     console.log(event.data)
 
     let payload = event.data.payload
+    if (payload.action === 'upload') {
```

Comment:
upload happens in popup.js , not contentscript.js, since the user adds the file in the extension popup.